File: 30 Days of Code/Ruby/Day15_LinkedList.rb
class Node

    attr_accessor :data,:next

  def initialize data
        @data = data
        @next = nil
    end
end

class Solution

    def insert(head,value)
      #Complete this method
      temp = Node.new(value)
      
        if head == nil
            return temp
        end 
        
        cur = head
        while cur.next
            cur = cur.next
        end
        cur.next = temp
        
        return head
    end

	def display(head)
       current = head
       while current 
          print current.data," "
          current = current.next
       end
   end
end
       
mylist= Solution.new
head=nil
T=gets.to_i
for i in 1..T
    data=gets.to_i
    head=mylist.insert(head,data)
end
mylist.display(head)

File: 30 Days of Code/Java/Day15_LinkedList.java
import java.io.*;
import java.util.*;

class Node {
	int data;
	Node next;
	Node(int d) {
        data = d;
        next = null;
    }
}

class Solution {

    public static  Node insert(Node head,int data) {
        //Complete this method
        Node temp = new Node(data);
        if(head == null)return temp;
        
        Node cur = head;
        while(cur.next!=null)cur = cur.next;
        
        cur.next = temp;
        return head;
    }

	public static void display(Node head) {
        Node start = head;
        while(start != null) {
            System.out.print(start.data + " ");
            start = start.next;
        }
    }

    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        Node head = null;
        int N = sc.nextInt();

        while(N-- > 0) {
            int ele = sc.nextInt();
            head = insert(head,ele);
        }
        display(head);
        sc.close();
    }
}